leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/LeshanServerBuilder.java

```diff
@@ -510,19 +512,10 @@ public LeshanServer build() {
                 dtlsConfigBuilder.setAdvancedCertificateVerifier(verifierBuilder.build());
             }
 
-            // Deactivate SNI by default
-            // TODO should we support SNI ?
-            if (incompleteConfig.isSniEnabled() == null) {
-                dtlsConfigBuilder.setSniEnabled(false);
-            }
-
             // Do no allow Server to initiated Handshake by default, for U device request will be allowed to initiate
             // handshake (see Registration.shouldInitiateConnection())
-            Boolean serverOnly = incompleteConfig.isServerOnly();
-            if (serverOnly == null || !serverOnly) {
-                if (incompleteConfig.getDefaultHandshakeMode() == null) {
-                    dtlsConfigBuilder.setDefaultHandshakeMode(DtlsEndpointContext.HANDSHAKE_MODE_NONE);
-                }
+            if (incompleteConfig.getDtlsRole() != DtlsRole.CLIENT_ONLY) {
+                dtlsConfigBuilder.set(DtlsConfig.DTLS_DEFAULT_HANDSHAKE_MODE, DtlsEndpointContext.HANDSHAKE_MODE_NONE);
```

Comment:
That was one place, where I didn't get the way the old implementation was.
The old way was: if the (server) endpoint is either in both roles, or not in the server role (=> therefore client role) then ...

Californium new Configuration:

```
	public String getDefaultHandshakeMode() {
		if (getDtlsRole() == DtlsRole.SERVER_ONLY) {
			return DtlsEndpointContext.HANDSHAKE_MODE_NONE;
		} else {
			return configuration.get(DtlsConfig.DTLS_DEFAULT_HANDSHAKE_MODE);
		}
	}
```

Californium old Configuration:
 
```
if (config.defaultHandshakeMode == null) {
	if (config.serverOnly) {
		config.defaultHandshakeMode = DtlsEndpointContext.HANDSHAKE_MODE_NONE;
	} else {
		config.defaultHandshakeMode = DtlsEndpointContext.HANDSHAKE_MODE_AUTO;
	}
}
 ```

maybe, we should clarify, what are the different cases and what is then wanted.

Reply:
I didn't think too much of this and just see that code seems not equals. 
Regarding this a little more maybe  the previous `|| !serverOnly ` does not make so much sense. 

>  what is then wanted.
The idea is in the comment : 
```
// Do no allow Server to initiated Handshake by default, for U device request will be allowed to initiate
// handshake (see Registration.shouldInitiateConnection())
```

The issue is that now there is no way to know if user set a value or not. 
This is a bit frustrating because a lot of the logic was thought around this idea.
That's even why we introduce the `getIncompleteConfig` so at first sight this looks like regression to me :disappointed:.   
I need to think more about this to be sure if this is a real problem or not.

Reply:
> The issue is that now there is no way to know if user set a value or not.

The point is, that the sequence has changed. Before, the user did something and afterward the defaults are amended, if the user didn't set it. Now, just set the intended defaults ahead, and pass the the initialized Configuration to the user in order to adapt it accordingly.

How I understand your idea now, I would just remove this lines here and instead, set the NONE ahead when the server Configurations are created.

Reply:
> The point is, that the sequence has changed. Before, the user did something and afterward the defaults are amended, if the user didn't set it. Now, just set the intended defaults ahead, and pass the the initialized Configuration to the user in order to adapt it accordingly.

I get it, hopping this will be a better sequence.

Before : 
1. user create config calling default constructor/Builder & set value he want to change. 
2. LeshanBuilder set default expected value for Leshan. 

Now : 
1. user must create a well initialized config adapter to Leshan. 
2. user then set value he want to change. 

I feel with the new way a bit more error prone as user can initialize the config with default californium value instead of Leshan one. And this could be not so easy to discover that you don't do it correctly. 

(Maybe there is a way to workaround with some WARN LOG or adding a key in the config at leshan initialization and check the key is present)

Reply:
I considered, that leshan offers some `createDefaultNetworkConfig`for the application types (e.g. lwm2m-client, lwm2m-data-server, lwm2m-bootstrap server) and users are starting to use them and add there configurations. use-case, user-configuration specific defaults are then not longer working, but I think, that usually it will take not that much time for a user, to get a "working adapted configuration".

Reply:
> Now :
> 
>     user must create a well initialized config adapter to Leshan.
>     user then set value he want to change.
> 

In the code it seems for me to be not that wired:

```
        // Create CoAP Config
        Configuration coapConfig;
        File configFile = new File(Configuration.DEFAULT_FILE_NAME);
        if (configFile.isFile()) {
            coapConfig = new Configuration();
            coapConfig.load(configFile);
        } else {
            coapConfig = LeshanClientBuilder.createDefaultNetworkConfig();
            // here comes the user setup
            // coapConfig.set(...)
            coapConfig.store(configFile);
        }
```

Reply:
>In the code it seems for me to be not that wired:

I don't get your point.

Reply:
> I don't get your point.

> I feel with the new way a bit more error prone

My intention was to show the code in order to see, what will be required.

Reply:
Ah OK get it now. 
So current concern is that user can missed to call `LeshanClientBuilder.createDefaultNetworkConfig();`

(so need to think more about this)